src/test/java/de/tum/in/www1/artemis/FileIntegrationTest.java

```diff
@@ -186,15 +184,14 @@ private FileUploadSubmission createFileUploadSubmissionWithRealFile() throws Exc
         FileUploadExercise fileUploadExercise = exerciseUtilService.findFileUploadExerciseWithTitle(course.getExercises(), "released");
         FileUploadSubmission fileUploadSubmission = ParticipationFactory.generateFileUploadSubmission(true);
         fileUploadSubmission = fileUploadExerciseUtilService.addFileUploadSubmission(fileUploadExercise, fileUploadSubmission, TEST_PREFIX + "student1");
-
         MockMultipartFile file = new MockMultipartFile("file", "file.png", "application/json", "some data".getBytes());
-        JsonNode response = request.postWithMultipartFile("/api/fileUpload?keepFileName=true", file.getOriginalFilename(), "file", file, JsonNode.class, HttpStatus.CREATED);
-        String responsePath = response.get("path").asText();
-        String filePath = fileService.manageFilesForUpdatedFilePath(null, responsePath,
-                FileUploadSubmission.buildFilePath(fileUploadExercise.getId(), fileUploadSubmission.getId()), fileUploadSubmission.getId(), true);
 
-        fileUploadSubmission.setFilePath(filePath);
-        return fileUploadSubmission;
+        userUtilService.changeUser(TEST_PREFIX + "student1");
+        FileUploadSubmission savedSubmission = request.postWithMultipartFiles("/api/exercises/" + fileUploadExercise.getId() + "/file-upload-submissions", fileUploadSubmission,
+                "submission", List.of(file), FileUploadSubmission.class, HttpStatus.OK);
+        userUtilService.changeUser(TEST_PREFIX + "instructor1");
```

Comment:
This line is apparently not needed. The two tests `testGetFileUploadSubmission()` and `testGetFileUploadSubmissionAsTutor()` that use this method also pass without changing the user back to instructor. I'm not sure what the point of the two tests even is, if even a student is able to access the submission through `request.get(fileUploadSubmission.getFilePath(), HttpStatus.OK, String.class);`
I think it would make sense to change the tests.

Reply:
It might work, but tests two different things:

I don't want to test if the submitting user can access it but if a user with elevated privileges can access it.

I changed it to a tutor though.